src/drivers/adc/adc.h

```diff
@@ -0,0 +1,17 @@
+#ifndef ADC_H
+#define ADC_H
+
+#include <stdbool.h>
+
+typedef enum {
+    PROCESSOR_BOARD_VOLTAGE = 0,
+} adc_channel_t;

```

Comment:
nit: imo any enum that might have more values added to it later should have an extra enum for the max value. example:
![image](https://github.com/user-attachments/assets/f3d483af-bedb-42b3-86dd-24ae1fda58f4)